internal/crypto/internal_ca.go

```diff
@@ -0,0 +1,160 @@
+package crypto
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"math/big"
+	"os"
+	"time"
+
+	oscrypto "github.com/openshift/library-go/pkg/crypto"
+)
+
+type internalCA struct {
+	Config          *TLSCertificateConfig
+	SerialGenerator oscrypto.SerialGenerator
+}
+
+func ensureInternalCA(certFile, keyFile, serialFile, subjectName string, expireDays int) (CABackend, bool, error) {
+	if ca, err := GetCA(certFile, keyFile, serialFile); err == nil {
+		return ca, false, err
+	}
+	ca, err := MakeSelfSignedCA(certFile, keyFile, serialFile, subjectName, expireDays)
+	return ca, true, err
```

Comment:
if this is an error the general expectation that the rest of the values are discarded. here it seems you want to pass true if its an error or not. is that expected